diff -ura --color httpdirfs-1.2.7.orig/src/link.c httpdirfs-1.2.7.new/src/link.c
--- httpdirfs-1.2.7.orig/src/link.c	2026-02-13 20:37:15.419465923 +0100
+++ httpdirfs-1.2.7.new/src/link.c	2026-02-13 20:39:01.890944706 +0100
@@ -271,7 +271,7 @@
  */
 static LinkTable *single_LinkTable_new(const char *url)
 {
-    char *orig_ptr = strrchr(url, '/') + 1;
+    const char *orig_ptr = strrchr(url, '/') + 1;
     char *ptr = curl_easy_unescape(NULL, orig_ptr, 0, NULL);
     LinkTable *linktbl = LinkTable_alloc(url);
     Link *link = Link_new(ptr, LINK_UNINITIALISED_FILE);
@@ -352,7 +352,7 @@
     /*
      * Check for stray '/' - Linkname should not have '/'
      */
-    char *slash = strchr(linkname, '/');
+    const char *slash = strchr(linkname, '/');
     if (slash) {
         int linkname_len = strnlen(linkname, MAX_FILENAME_LEN) - 1;
         if (slash - linkname != linkname_len) {
diff -ura --color httpdirfs-1.2.7.orig/src/sonic.c httpdirfs-1.2.7.new/src/sonic.c
--- httpdirfs-1.2.7.orig/src/sonic.c	2026-02-13 20:37:15.429629059 +0100
+++ httpdirfs-1.2.7.new/src/sonic.c	2026-02-13 20:38:29.189775716 +0100
@@ -235,7 +235,7 @@
             /*
              * Skip to the last '/' if it exists
              */
-            char *s = strrchr(attr[i + 1], '/');
+            const char *s = strrchr(attr[i + 1], '/');
             if (s) {
                 strncpy(link->linkname, s + 1, MAX_FILENAME_LEN);
             } else {
